render_pass.py

```diff
@@ -0,0 +1,111 @@
+
+update_render_passes_original = cycles.CyclesRender.update_render_passes
+render_original = cycles.CyclesRender.render
+del_original = cycles.CyclesRender.__del__
+
+def register_render_pass():
+    def update_render_passes_decorator(original):
+        def update_render_passes(self, scene=None, renderlayer=None):
+            result = original(self, scene, renderlayer)
+            self.register_pass(scene, renderlayer, "Dream Textures", 4, "RGBA", 'COLOR')
+            return result
+        return update_render_passes
+    cycles.CyclesRender.update_render_passes = update_render_passes_decorator(cycles.CyclesRender.update_render_passes)
+    def render_decorator(original):
+        def render(self, depsgraph):
+            scene = depsgraph.scene if hasattr(depsgraph, "scene") else depsgraph
+            if not scene.dream_textures_render_properties_enabled:
+                return original(self, depsgraph)
+            result = original(self, depsgraph)
+            try:
+                # TODO: Create the render result from scratch
+                original_result = self.get_result()
+                self.add_pass("Dream Textures", 4, "RGBA")
+                scale = scene.render.resolution_percentage / 100.0
+                size_x = int(scene.render.resolution_x * scale)
+                size_y = int(scene.render.resolution_y * scale)
+                render_result = self.begin_result(0, 0, size_x, size_y)
+                for original_layer in original_result.layers:
+                    layer = None
+                    for layer_i in render_result.layers:
+                        if layer_i.name == original_layer.name:
+                            layer = layer_i
+                    for original_render_pass in original_layer.passes:
+                        render_pass = None
+                        for pass_i in layer.passes:
+                            if pass_i.name == original_render_pass.name:
+                                render_pass = pass_i
+                        if render_pass.name == "Dream Textures":
+                            self.update_stats("Dream Textures", "Starting")
+                            def image_callback(event, shared_memory_name, seed, width, height, upscaled=False):
+                                self.update_stats("Dream Textures", "Pushing to render pass")
+                                # Only use the non-upscaled texture, as upscaling is currently unsupported by the addon.
+                                if not upscaled:
+                                    print(width, height)
+                                    print(len(render_pass.rect))
+                                    shared_memory = SharedMemory(shared_memory_name)
+                                    pixels = np.frombuffer(shared_memory.buf, dtype=np.float32)
+                                    reshaped = pixels.reshape((width * height, 4))
+                                    render_pass.rect.foreach_set(reshaped)
```

Comment:
This can error and freeze Blender with `TypeError: expected sequence size 133120, got 131072` when the output properties resolution isn't 512 or less in X or Y and when X or Y isn't a multiple of 64. Suggested changes in #245